Frameworks/Starboard/EbrFile.cpp

```diff
@@ -0,0 +1,273 @@
+    fileToAdd = EbrIOFile::CreateInstance(file, mode, share, pmode);
+    return (fileToAdd) ? EbrFile::AddFile(std::move(fileToAdd)) : -1;
+}
+
+int EbrClose(int fd) {
+    EbrFile::RemoveFile(fd);
+    return 0;
+}
+
+int EbrFstat(int fd, struct stat* ret) {
+    return EbrFile::GetFile(fd)->Stat(ret);
+}
+
+int EbrFstat64i32(int fd, struct _stat64i32* ret) {
+    return EbrFile::GetFile(fd)->Stat64i32(ret);
+}
+
+intptr_t EbrGetOSFHandle(int fd) {
+    return EbrFile::GetFile(fd)->GetOSFHandle();
+}
+
+int EbrRead(int fd, void* dest, size_t count) {
+    return EbrFile::GetFile(fd)->Read(dest, count);
+}
+
+int EbrWrite(int fd, const void* src, size_t count) {
+    return EbrFile::GetFile(fd)->Write(src, count);
+}
+
+int EbrLseek(int fd, __int64 pos, int whence) {
+    return EbrFile::GetFile(fd)->Lseek(pos, whence);
+}
+
+int EbrTruncate64(int fd, __int64 size) {
+    return EbrFile::GetFile(fd)->Truncate64(size);
+}
+
+size_t EbrTell(int fd) {
+    return EbrFile::GetFile(fd)->Tell();
+}
+
+int EbrFflush(int fd) {
+    return EbrFile::GetFile(fd)->Flush();
+}
+
+bool EbrRemoveEmptyDir(const char* path) {
+    return RemoveDirectoryA(CPathMapper(path));
+}
+
+bool EbrRename(const char* path1, const char* path2) {
+    return rename(CPathMapper(path1), CPathMapper(path2)) == 0;
+}
+
+bool EbrUnlink(const char* path) {
+    return _unlink(CPathMapper(path)) == 0;
+}
+
+#define FSROOT "."
+#define mkdir _mkdir
+char g_WritableFolder[2048] = ".";
+
+void EbrSetWritableFolder(const char* folder) {
+    strcpy_s(g_WritableFolder, folder);
+}
+
+const char* EbrGetWritableFolder() {
+    return g_WritableFolder;
+}
+
+bool EbrGetRootMapping(const char* dirName, char* dirOut, uint32_t maxLen) {
+    if (dirName == NULL) {
+        strcpy_s(dirOut, maxLen, FSROOT);
+        return true;
+    }
+    if (_stricmp(dirName, "Documents") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\Documents", g_WritableFolder);
+        mkdir(dirOut);
+
+        char tmpDir[4096];
+        strcpy_s(tmpDir, dirOut);
+        strcat_s(tmpDir, "\\Library");
+        mkdir(tmpDir);
+        return true;
+    }
+    if (_stricmp(dirName, "Cache") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\cache", g_WritableFolder);
+        mkdir(dirOut);
+        return true;
+    }
+    if (_stricmp(dirName, "Library") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\Library", g_WritableFolder);
+        mkdir(dirOut);
+        return true;
+    }
+    if (_stricmp(dirName, "AppSupport") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\AppSupport", g_WritableFolder);
+        mkdir(dirOut);
+        return true;
+    }
+    if (_stricmp(dirName, "tmp") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\tmp", g_WritableFolder);
+        mkdir(dirOut);
+        return true;
+    }
+    if (_stricmp(dirName, "shared") == 0) {
+        sprintf_s(dirOut, maxLen, "%s\\shared", g_WritableFolder);
+        mkdir(dirOut);
+        return true;
+    }
+    static std::regex drive("[a-zA-Z]:");
```

Comment:
>regex [](start = 16, length = 5)

I know this isn't your code, but we're using a whole regex engine to detect a single letter and a colon??!!!??